doc/design.rst

```diff
@@ -606,3 +597,59 @@ are deleted, the particular snapshot vanished and all snapshots
 depending on data that has been added in the snapshot cannot be restored
 completely. Restic is not designed to detect this attack.
 
+With the aforementioned assumptions and guarantees in mind, the following are
+examples of things an adversary could achieve in various circumstances.
+
+An adversary with read access to your backup storage location could: 
+- Attempt a brute force password guessing attack against a copy of the 
+repository (even more reason to use long, random passwords).
+
+An adversary with write access to your files at the storage location could: 
+- Delete or manipulate your backups, thereby impairing your ability to restore
+files from the compromised storage location. 
+- Delete specific files (by using timestamps to determine which files belong to
+what snapshots), resulting in snapshots vanishing and preventing all dependent 
+snapshots from being completely restored.
+
+An adversary who compromises (via malware, physical access, etc.) the host
+system making backups could: 
+- Render the entire backup process untrustworthy (e.g., intercept password, 
+copy files, manipulate data).
+
+An adversary with the ability to add data to the repo (but not remove something)
+could: 
+- upload pack files with wrong headers, but only if the attacker knows the 
+password to encrypt the pack. For example, an attacker could include a header 
+saying the file contains a blob with a specific hash when that's not really the
+case. 
+- upload an index file claiming a specific blob IDs exist in the repository, 
+but only if the attacker knows the password. In this case, restic will not 
+question this information until the index is rebuilt (by rebuild-index or 
+prune). 
+- if forget with time-based policies is regularly run against the repository 
+(particularly when automated), an attacker could introduce new snapshots (that
+are empty or contain garbage data) with specifically-crafted timestamps in an 
+attempt to get a future forget operation to remove some or all existing good 
+snapshots from the repository. However, this requires the attacker to know the
+repository password or master key in order to encrypt the malicious snapshot 
+objects so that forget will be able to read them.
+
+An adversary with network access could: 
+- attempt to DoS the restic server.
```

Comment:
I see your point about "restic server" being ambiguous. Even with your suggested revision, I'm struggling how to better clarify this one.  DoS is a threat, but a very nuanced one. 

Another scenario/nit is cases when the "server storing the backup repository" is a cloud service like S3. Your average adversary isn't going to have any luck performing a DoS attack against that. The attacker would have better luck going after the endpoints attempting to upload data to the restic repo. 

I also somewhat disagree with the local backup exception. The "adversary with network access" phrase implies this doesn't apply when storing locally. Also, if an attacker can DoS a box (regardless if restic itself is network exposed to receive backups), it could cause collateral damage (albeit the DoS would not be against the restic service, but the infrastructure it runs on). I don't know that any of this is worth calling out explicitly though. 

This is a long way of saying I need help thinking through how to better explain the DoS threat.

Reply:
The "attempt to" should mostly cover the part that the attacker quite likely won't be able to bring down S3. You could add that the attacker could DoS the network connection between client and server. That should be generic enough to DoS attacks on intermediate nodes or the client itself. Intermediate node or the client usually provide no external services, thus an attacker would have to resort to flooding the networking connection with too many requests.